app/controllers/api/projects_controller.rb

```diff
@@ -67,9 +67,16 @@ def updated
 
   def dependencies
     subset = params.fetch(:subset, "default")
-    project_json = find_project_as_json_with_dependencies!(params[:platform], params[:name], params[:version], subset)
 
-    render json: project_json
+    if params[:v2] == "true"
+      @project = Project.find_best!(params[:platform], params[:name], [:repository, :versions])
+      @version = @project.find_version!(params[:version])
```

Comment:
Another pretty large optimization here would be this https://github.com/librariesio/libraries.io/blob/100cafcc0bae517dd43b15ea51a7ba00fca7bd7f/app/models/project.rb#L599 where we grab and hydrate all versions to then throw them all out.

Reply:
Ooh, that's a good find.

Reply:
Actually, this one might be tricky to do now, and we'll use the `versions` associations in the JB template anyway so I'm gonna look at this later.

Reply:
We only use them for the non-limited version